File: python/structures/src/sorts/selection.py
from structures.src.util.sort_util import swap

def sort(arr):
    out = list(arr)

    for i in range(0, len(out)):
        min_index = i

        for j in range(i, len(out)):
            if(out[j] < out[min_index]):
                min_index = j

        swap(out, i, min_index)

    return out

File: c/sorts/selection.c
#include "../util/swap.h"

void Selection_sort(int *arr, int length) {
    for(int i = 0; i < length; i++) {
        int min = i;

        for(int j = i + 1; j < length; j++) {
            if(arr[j] < arr[min]) {
                min = j;
            }
        }

        swap(arr, i, min);
    }
}
